### PR TITLE
Enable nice imports from javascript in typescript

### PR DESCRIPTION
**Pull request by molly-moen**
Previously, in order to import our own javascript functions or components into typescript, we had to use the following syntax, which is outdated and also does not provide nice intellisense:
```
const UserPreferences = require('@cdo/apps/lib/util/UserPreferences');
```
With this change, we can use the nicer, more helpful syntax of:
```
import UserPreferences from '@cdo/apps/lib/util/UserPreferences';
```

In order to enable this, we needed to parse our javascript files as part of the typescript compilation step. There were a few options for this: parse every *.js and *.jsx file under `/src`, parse only the folders of files we are currently importing, or specify the exact files we want to include. The more files we add for the typescript compiler to parse, the longer our `yarn build` and `yarn start` takes (interestingly, rebuilding under `yarn start` does not seem to be affected). If we include all files, the build time increases by 50 seconds. If we include just specific folders, the build time increases by 24 seconds. If we only include specific files (and all of `/music` since it is transitioning from js to ts) the build time increases by 10 seconds. However, with each level of strictness, the likelihood increases that a developer will need to know to update `tsconfig` to include a new javascript file.

Since the most important metric for developer productivity is rebuild time, I decided to go with the middle option, which increases our overall build time by around 24 seconds. It saves us some overall build time, but the need for developers to update `tsconfig` is much lower. 

As I was testing this, I found some invalid html in `NewUnitForm` which `tsc` complained about. I checked with [teacher-tools](https://codedotorg.slack.com/archives/C045UAX4WKH/p1684364165673599) and it was safe to remove.
## Links

- jira ticket: [SL-791](https://codedotorg.atlassian.net/browse/SL-791)

## Testing story
Mostly tested this by running the build script. The main work here was figuring out how to get `allowJs` in `tsconfig` to work--if we didn't add the javascript files to parse, we got build errors. Once I added the js files to `include`, the build worked again.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/tsconfig.json [hunk 1]
```diff
@@ -1,4 +1,18 @@
 {
+  "include": [
+    "./src/**/*.ts",
+    "./src/**/*.tsx",
+    // Include all folders with .js and .jsx files that ts files reference.
+    // If you need to import a .js or .jsx file into a .ts or .tsx file, you
+    // can add its folder here to enable the syntax:
+    // import {component} from "folder/component";
```
Comment (sanchitmalhotra126) [inline apps/tsconfig.json:8]: Should we also add this to apps/README.md for reference? (maybe we can create a small TypeScript migration section that we can continue to fill out later?)

**Review by sanchitmalhotra126 — APPROVED** [review]
🎉 thanks for tracking this down!

File: apps/tsconfig.json (hunk 1 above)
Comment (molly-moen) [reply apps/tsconfig.json:8]: oh good call

**Review by fisher-alice — APPROVED** [review]
Thanks for answering my questions about this PR in [this Slack thread](https://codedotorg.slack.com/archives/C0T0PNR0D/p1684429791196119)!

File: apps/README.md [hunk 2]
```diff
@@ -220,6 +220,16 @@ To add a new package using npm, e.g., `lodash`, run: `npm i --save-dev lodash`
 - `--save-dev` adds the dependency to node's package.json, freezing the current version
 - Because the build process is done in dev mode, include dependencies as devDependencies rather than production dependencies
 
+### Typescript Migration
+We are trying out Typescript in our repository, and currently have a combination of Typescript and Javascript files. Typescript files can be added anywhere in `/src`, and wil
```
Comment (breville) [inline apps/README.md:224]: super nit: `wil`

**Pull request merged**